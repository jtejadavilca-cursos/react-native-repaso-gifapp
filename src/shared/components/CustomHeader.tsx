interface Props {
    title: string;
    description?: string;
}

export const CustomHeader = ({ title, description }: Props) => {
    return (
        <div className="content-center">
            <h1>{title}</h1>

            {description && <p data-testid="description">{description}</p>}
        </div>
    );
};
