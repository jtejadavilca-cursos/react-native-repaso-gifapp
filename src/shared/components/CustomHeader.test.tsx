import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { CustomHeader } from "./CustomHeader";

describe("CustomHeader", () => {
    const testTitle = "Title test";
    test("Should render component properly", () => {
        const { container } = render(<CustomHeader title={testTitle} />);
        expect(container).toMatchSnapshot();
    });

    test("Should not render a title - Way 1", () => {
        render(<CustomHeader title={testTitle} />);

        expect(screen.getByRole("heading").innerHTML).toContain(testTitle);
        expect(screen.queryByTestId("description")).toBeNull();
    });

    test("Should not render a title - Way 2", () => {
        const { container } = render(<CustomHeader title={testTitle} />);

        const divElement = container.querySelector(".content-center");
        const h1 = divElement?.querySelector("h1");
        expect(h1?.innerHTML).toBe(testTitle);
        expect(container.innerHTML).toContain(testTitle);

        const p = divElement?.querySelector("p");
        expect(p).toBeNull();
    });

    test("Should render a title", () => {
        const descriptionText = "Test description";
        render(<CustomHeader title={testTitle} description={descriptionText} />);

        expect(screen.getByRole("heading").innerHTML).toContain(testTitle);
        expect(screen.getByText(descriptionText)).toBeDefined();
        expect(screen.getByRole("paragraph")).toBeDefined();
        expect(screen.queryByTestId("description")).toBeDefined();
        expect(screen.getByTestId("description").innerHTML).toContain(descriptionText);
    });
});
