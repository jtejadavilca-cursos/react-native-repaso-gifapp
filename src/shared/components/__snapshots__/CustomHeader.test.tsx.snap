// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomHeader > Should render component properly 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Title test
    </h1>
  </div>
</div>
`;
